### PR TITLE
S2211

### PR DESCRIPTION
**Pull request by yeunga**
Removed code to reopen file handle from args.obs_file and updated unit tests.
Added code to close args.obs_file
Made code more robust by moving client.get_observation() in multiprocess_observation_id() to try...catch block.
Corrected help text.

### CODE REVIEW

File: caom2repo/README.rst [hunk 1]
```diff
@@ -1,10 +1,6 @@
 =========
 Caom2Repo
 =========
-
-.. image:: https://img.shields.io/pypi/v/caom2repo.svg   
-    :target: https://pypi.python.org/pypi/caom2repo
-
 Client caom2-repo
```
Comment (andamian) [inline caom2repo/README.rst]: Please add this back. It displays an icon with the version number published on pypi.

File: caom2repo/caom2repo/core.py [hunk 2]
```diff
@@ -602,14 +602,24 @@ def multiprocess_observation_id(collection, observationID, plugin, subject,
...
                 " method of your plugin.".format(str(e)))
+        else:
+            # other unexpected TypeError
+            raise e
     except Exception as e:
-        failed = observation.observation_id
-        rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
+        if observation is None:
+            rootLogger.error('Failed to get observation using observationID = {} - Reason: {}'.format(observationID, e))
+            raise e
```
Comment (andamian) [inline caom2repo/caom2repo/core.py]: Maybe it should raise e only with halt on error? And move outside the if/case at the end.

File: caom2repo/caom2repo/core.py [hunk 3]
```diff
@@ -602,14 +602,24 @@ def multiprocess_observation_id(collection, observationID, plugin, subject,
...
     except KeyboardInterrupt as e:
         # user pressed Control-C or Delete
-        rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
+        if observation is None:
+            rootLogger.error('Failed to get observation using observationID = {} - Reason: {}'.format(observationID, e))
+        else:
+            rootLogger.error('FAILED {} - Reason: {}'.format(observation.observation_id, e))
         raise e
 
     visited = observation.observation_id
```
Comment (andamian) [inline caom2repo/caom2repo/core.py]: If exception is not re-raised when get_authorization fail, the code will end up here and observation will be None. Safer to replace visited = observationID

Comment (yeunga) [reply caom2repo/caom2repo/core.py]: This block of code is only executed when there is a KeyboardInterrupt, no? visited should only be set when there is no exception/error, no?

Comment (andamian) [reply caom2repo/caom2repo/core.py]: We increment visited when errors occur on post_observation, don't we. Visited is the total and includes errors or skips.

**Pull request merged**